src/System.Linq.Expressions/tests/BinaryOperators/Arithmetic/BinaryAddTests.cs

```diff
@@ -282,6 +336,54 @@ private static void VerifyShortAdd(short a, short b)
             }
         }
 
+        private static void VerifyShortAddOvf(short a, short b)
```

Comment:
> I am not sure the C# part of validation like "csResult = checked((short)(a + b));"

That's why I was suggesting it also take a delegate, e.g. (a, b) => checked((short)(a + b)).